lib/src/main/java/com/diffplug/spotless/npm/RoundtrippableFile.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2024 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.npm;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import javax.annotation.Nonnull;
+
+import com.diffplug.spotless.ThrowingEx;
+
+/**
+ * Wrapper for a file that conforms to the following idea:
+ * 1. Equals and hashcode should not include absolute paths and such - optimize for buildcache keys
+ * 2. Serialized/deserialized state can include absolute paths and such and should recreate a valid/runnable state
+ */
+class RoundtrippableFile implements Serializable {
```

Comment:
This is inspired by `FileSignature` - except that it does not use the serializable-for-equality-hack. 
Maybe we could combine these two classes?